manual/core/integration/README.md

```diff
@@ -0,0 +1,418 @@
+[configuration reference](../configuration/reference/) (look for "exposed as a Timer").
+
+If all of these metrics are disabled, you can remove the dependency:
+
+```xml
+<dependency>
+  <groupId>com.datastax.oss</groupId>
+  <artifactId>java-driver-core</artifactId>
+  <version>4.0.0-beta1</version>
+  <exclusions>
+    <exclusion>
+      <groupId>org.hdrhistogram</groupId>
+      <artifactId>HdrHistogram</artifactId>
+    </exclusion>
+  </exclusions>
+</dependency>
+```
+
+#### Documenting annotations
+
+The driver team uses annotations to document certain aspects of the code:
+
+* thread safety with [Java Concurrency in Practice](http://jcip.net/annotations/doc/index.html)
+  annotations `@Immutable`, `@ThreadSafe`, `@NotThreadSafe` and `@GuardedBy`;
+* nullability with [SpotBugs](https://spotbugs.github.io/) annotations `@Nullable` and `@NonNull`.
+
+This is mostly used during development; while these annotations are retained in class files, they
+serve no purpose at runtime. Whether to make them optional dependencies is up for debate:
+
+* if they are required, it's two additional JARs that every client has to pull in (admittedly they
+  are quite small);
+* if they are optional, the bytecode will reference missing classes. This is not a blocker, but it
+  can manifest to the end user in a few ways:
+  
+    * if you navigate to the driver's sources in your IDE, the missing annotations will be
+      highlighted as errors (note however that modern IDEs such as IntelliJ IDEA can analyze
+      nullability annotations even if they are missing from the classpath); 
+    * this produces compiler warnings (see [this discussion][guava] of a similar issue for Google's
+      Guava library).
+
+The Java driver team has decided to make the dependencies optional. If that creates any problem for
+you, the workaround is to redeclare them explicitly in your application:
+
+```xml
+<dependency>
+  <groupId>com.datastax.oss</groupId>
+  <artifactId>java-driver-core</artifactId>
+  <version>4.0.0-beta1</version>
+</dependency>
+<dependency>
+  <groupId>com.github.stephenc.jcip</groupId>
+  <artifactId>jcip-annotations</artifactId>
+  <version>1.0-1</version>
+</dependency>
+<dependency>
+  <groupId>com.github.spotbugs</groupId>
+  <artifactId>spotbugs-annotations</artifactId>
+  <version>3.1.3</version>
+</dependency>
+```
+
+#### Mandatory dependencies
+
+The remaining core driver dependencies are the only ones that are truly mandatory:
+
+* the [native protocol](https://github.com/datastax/native-protocol) layer. This is essentially part
+  of the driver code, but was externalized for reuse in other projects;
+* `java-driver-shaded-guava`, a shaded version of [Guava](https://github.com/google/guava). It is
+  relocated to a different package, and only used by internal driver code, so it should be
+  completely transparent to third-party code;
+* the [SLF4J](https://www.slf4j.org/) API for [logging](../logging/).
```

Comment:
I had never thought specifically about this, but it's pretty cool: you can run the driver with a minimal set of 3 dependencies, two of them being our internal stuff. SLF4J is the only one that could clash with 3rd-party code.